src/obstacle_params_parser.cpp

```diff
@@ -0,0 +1,133 @@
+#pragma once
+
+#include "panda_torque_mpc/obstacle_params_parser.h"
+
+#include <Eigen/Dense>
+#include <boost/shared_ptr.hpp>
+#include <fstream>
+#include <memory>
+#include <vector>
+
+#include <pinocchio/fwd.hpp>
+#include <pinocchio/multibody/model.hpp>
+
+#include <ros/ros.h>
+#include <yaml-cpp/yaml.h>
+
+namespace panda_torque_mpc {
+
+  ObstacleParamsParser::ObstacleParamsParser(
+      const std::shared_ptr<ros::NodeHandle> &nh,
+      const boost::shared_ptr<pinocchio::Model> &pin_model,
+      const boost::shared_ptr<pinocchio::GeometryModel> &collision_model)
+      : nh_(nh), pin_model_(pin_model), collision_model_(collision_model){};
+
+  ObstacleParamsParser::~ObstacleParamsParser() {};
+
+void ObstacleParamsParser::addCollisions()
+{
+  // First adding the obstacles to the geometry model
+ int obs_idx = 0;
+ while (nh_->hasParam("obstacle_" + std::to_string(obs_idx))) {
+
+  std::string obstacle_name;
+
+  obstacle_name = "obstacle_" + std::to_string(obs_idx);
+  std::string type;
+
+  nh_->getParam(obstacle_name + "/type", type);
+
+  Eigen::VectorXd pose;
+  nh_->getParam(obstacle_name + "/pose", pose);
+
+  switch (type) {
+    case "sphere":
+      double radius;
+      nh_->getParam(obstacle_name + "/radius", radius);
+      addSphere(obstacle_name, radius, pose);
+      break;
+
+    case "box":
+      double x; double y; double z;
+      nh_->getParam(obstacle_name + "/x", x);
+      nh_->getParam(obstacle_name + "/y", y);
+      nh_->getParam(obstacle_name + "/z", z);
+      addBox(obstacle_name, x, y, z, pose);
+      break;
+
+    case "capsule":
+      double radius; 
+      double halfLength;
+      nh_->getParam(obstacle_name + "/radius", radius);
+      nh_->getParam(obstacle_name + "/halfLength", halfLength);
+      addCapsule(obstacle_name, radius, halfLength, pose);
+      break;
+    }
+  obs_idx += 1;
+ }
+  // Adding the collision pairs to the geometry model
+
+  std::vector<std::vector<std::string>> collision_pairs;
+  nh_->getParam("collision_pairs", collision_pairs);
+  
+  for (std::vector<std::string> collision_pair : collision_pairs) {
```

Comment:
```suggestion
  for (const auto& collision_pair : collision_pairs) {
```
This is a more popular syntax. Additionally, it is faster to execute as it is using references to the variables